lib/tools/adb-commands.js

```diff
@@ -1528,10 +1528,19 @@ methods.setHttpProxy = async function (proxyHost, proxyPort) {
     throw new Error(`Call to setHttpProxy method with undefined proxy_port ${proxy}`);
   }
   await this.setSetting('global', 'http_proxy', proxy);
-  await this.setSetting('secure', 'http_proxy', proxy);
-  await this.setSetting('system', 'http_proxy', proxy);
-  await this.setSetting('system', 'global_http_proxy_host', proxyHost);
-  await this.setSetting('system', 'global_http_proxy_port', proxyPort);
+  await this.setSetting('global', 'global_http_proxy_host', proxyHost);
+  await this.setSetting('global', 'global_http_proxy_port', proxyPort);
+};
+
+/**
+ * Delete HTTP proxy in device global settings.
+ * Rebooting the test device is necessary to apply the change.
+ */
+methods.deleteHttpProxy = async function () {
+  await this.shell(['settings', 'delete', 'global', 'http_proxy']);
+  await this.shell(['settings', 'delete', 'global', 'global_http_proxy_host']);
+  await this.shell(['settings', 'delete', 'global', 'global_http_proxy_port']);
+  await this.shell(['settings', 'delete', 'global', 'global_http_proxy_exclusion_list']);
```

Comment:
where do we set exclusion list property?

Reply:
Maybe, we can set it via adb command with `,` separation explained:
https://android.googlesource.com/platform/frameworks/base/+/android-9.0.0_r21/core/java/android/provider/Settings.java#10376

When I set `http_proxy` via adb command, `global_http_proxy_exclusion_list=` (the value was blank) was also generated. So, I deleted here to clean settings generated by `http_proxy`.
Let me leave a comment

Reply:
I'll add it in `setHttpProxy`

Reply:
an explanation comment is pretty enough